engine/kv_engine.cpp

```diff
@@ -540,18 +541,6 @@ Status KVEngine::RestoreSkiplistHead(DLRecord* pmem_record,
         string_view_2_string(pmem_record->Key()).c_str());
     return Status::Abort;
   }
-  Skiplist* skiplist;
-  {
-    std::lock_guard<std::mutex> lg(list_mu_);
-    skiplists_.push_back(std::make_shared<Skiplist>(
-        pmem_record, name, id, comparator, pmem_allocator_, hash_table_));
-    skiplist = skiplists_.back().get();
-    if (configs_.opt_large_sorted_collection_restore) {
-      sorted_rebuilder_->AddRecordForParallelRebuild(
-          pmem_allocator_->addr2offset(pmem_record), false, nullptr);
-    }
-  }
-  compare_excange_if_larger(list_id_, id + 1);
 
   // Here key is the collection name
   auto hint = hash_table_->GetHint(name);
```

Comment:
We have multiple versions of SkiplistRecord now. Should we check their linkage and repair them?